### ISSUE TITLE
E2E Tests for WP nightly consistently fail

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#1710]
## Bug Description

Our E2E test job which runs on the nightly/trunk version of WordPress consistently fails with the same error (a timeout in the Search Console date range test).

This job is allowed to fail as it is running on a pre-release version of WordPress but the purpose of this job is also to address failures before the upcoming release if there are breaking changes to be addressed.

Example failing jobs:
* https://travis-ci.com/github/google/site-kit-wp/jobs/352990831
* https://travis-ci.com/github/google/site-kit-wp/jobs/352930516
* https://travis-ci.com/github/google/site-kit-wp/jobs/352855407

Note: the PHP tests job which fails running on WP nightly/trunk is failing for a different reason related to the nightly _PHP version_ used in that job, so this job is expected to continue to fail for the time being.

## Screenshots

![image](https://user-images.githubusercontent.com/1621608/85519223-d6a01e00-b609-11ea-9553-92be34ce22cd.png)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The **E2E Tests (WordPress nightly)** job should pass

## Implementation Brief

In WP 5.5, there is a change that causes `esc_url_raw( false )` to return an empty string instead of `false` (I'm not sure exactly where but it is reproducible). This causes references to `_googlesitekitLegacyData.permaLink` used in data requests to calculate a different cache key for the request which doesn't match what we've provided causing the test to fail because the component is in a "waiting for data" state.

```
# WP 5.4.2
wp> esc_url_raw( false )
=> bool(false)
wp> exit
www-data@2efd5ac3af93:~/wordpress$ wp core update --version=nightly
Updating to version nightly (en_US)...
Downloading update from https://wordpress.org/nightly-builds/wordpress-latest.zip...
Unpacking the update...
Cleaning up files...
No files found that need cleaning up.
Success: WordPress updated successfully.
www-data@2efd5ac3af93:~/wordpress$ wp shell
wp> esc_url_raw( false )
=> string(0) ""
```

* Update `permaLink` to always be `false` if not set https://github.com/google/site-kit-wp/blob/7442f4add1ba2b79449aabb2e4b5283949bf0f43/includes/Core/Assets/Assets.php#L816
* Merge #1837 

I also diffed `_googlesitekitLegacyData` between 5.4.2 and the current nightly and found that other than `timestamp` this was the only change, which is expected.

## QA Brief

* The E2E tests for the `E2E=1 WP_VERSION=nightly` job should pass along with the other jobs for Travis builds after #1837 was merged

Note there is still some instability in the E2E tests which has another issue to address it (see #1821) so check that any failures are not related to these. 

## Changelog entry

* N/A

**Comment by tofumatt** [comment]
IB ✅

**Comment by tofumatt** [comment]
Code review is good, just assigning to @aaemnnosttv to merge.

**Comment by cole10up** [comment]
This is verified fixed:

https://travis-ci.com/github/google/site-kit-wp/builds/177735168

Passed QA ✅
